ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/AbstractRecord.java

```diff
@@ -473,9 +473,21 @@ public static AbstractRecord create (int type)
         }
 	    catch (final Throwable ex)
 	    {
-	        ex.printStackTrace();
-	        
-	        return null;
+			try {
+				Class rt = RecordType.typeToClass(type);
+				String className = (rt == null) ? "null" : rt.getName();
+				int mb = 1024 * 1024;
```

Comment:
My general impression is that the memory consumpsion data is a specific entry which is not meant for a general usage. It's more about to discover an issue. But issue discovery should be processed with a dev branch not on master.
It would be nice to change the message to log WARN instead of exception being printed on stdout. And information about what is the record type is really useful. But I don't think the same about the memory calculation.

Reply:
> My general impression is that the memory consumpsion data is a specific entry which is not meant for a general usage. It's more about to discover an issue. But issue discovery should be processed with a dev branch not on master.

Disagree, the message is providing more than what was already there (which was just an exception trace). This code path is almost never seen so we need something that will appear under general testing conditions.

> It would be nice to change the message to log WARN instead of exception being printed on stdout. 

Disagree, the message is already printed at WARN level (have you read the rest of the code immediately following this line).

> And information about what is the record type is really useful.

Disagree, the message is already printing the record type.

> But I don't think the same about the memory calculation.

Disagree, This code path is almost never seen so we need something that will appear under general testing conditions.

Reply:
I probably haven't expressed my ideas with the right words. I will try that better, sorry for that.
I meant it in way: The state of code before was printing stacktrace to stdout. Then using logger and WARN with printing the record type is nice and it's needed.
Still I think printing memory consumption into the WARN message is not right.

Reply:
I agree with Ondra's summary, it seems usual to add in the memory to the tracing. Perhaps there is a reason that this is needed for this case and in that situation it may well be warranted. Per default I was expecting the new statement to just be something like:
```
Class rt = RecordType.typeToClass(type);
String className = (rt == null) ? "null" : rt.getName();
tsLogger.i18NLogger.warn_coordinator_AbstractRecord_create_failure(type, className, ex);
```
It's not clear how the memory information helps here from my point of view but as I say if there is some reason to add it we can do so, though by the nature of the volatilty of the JVM the numbers for the memory that we log could well be different at the time we catch the exception to the time when the event that the exception was raised were.

Reply:
I added it because we have no idea about why we get the InstantiationException and printing some data about memory usage may be useful.

Reply:
IMO I think that is too speculative. Perhaps you could separately trace the message but even that seems speculative. It could be a problem with diskspace for example?

Reply:
None of the standard reasons for instantiation exceptions explain what we see. The mem usage data may have given a hint as to whether or not the JVM is stressed. But I have removed the extra data in my last commit.